### ISSUE TITLE
Receiving a 403 HTTP Response when making a request with no User-Agent in header

### ISSUE THREAD

**Issue by lodenrogue** [phalt/pokeapi#135]
Getting this: "Server returned HTTP response code: 403 for URL: http://pokeapi.co/api/v2/encounter-method/old-rod" when making an api call without a User-Agent in the header. I found a work around on my end but about a week ago this was working without the need of a User-Agent.

**Comment by phalt** [comment]
@lodenrogue your http client should definitely be setting a User-Agent heading, which one are you using?

**Comment by lodenrogue** [comment]
I'm using the Jackson library in Java. The default implementation doesn't set headers but there is an option to pass a Reader so I added the header there. I'm just curious as to why it was working a week ago but not now. Was something changed?

**Comment by phalt** [comment]
@lodenrogue we did a bunch of upgrades to a few packages, including Django and Django REST Framework, so it could have been part of a security fix.

**Comment by lodenrogue** [comment]
Makes sense. Thank you.

**Comment by martinpeck** [comment]
I'm seeing the same thing, but with a particular User-Agent header. 
If I make a call to the API like this...

```python
data = urllib.request.urlopen("http://pokeapi.co/api/v1/pokemon/"+str(num)).readall()
```

... then urllib sets the User-Agent to `Python-urllib/3.5` and I see a 403. 

If I sent the User-Agent to anything else, it's fine.

For example, this works:

```python
url = "http://pokeapi.co/api/v1/pokemon/"+str(num)
request = urllib.request.Request(url)

request.add_header('User-Agent',"cheese")

data = urllib.request.urlopen(request).read()
```

I've created a Gist to show the issue I'm seeing...

https://gist.github.com/martinpeck/2e2437bdc586a74cbc9e

**Comment by phalt** [comment]
@martinpeck ah ha! You found the GitHub issue :) (I just sent you an email about this). That Gist will be helpful. It appears we're not setting the ALLOWED_HOSTS or a similar DRF setting somewhere.

**Comment by martinpeck** [comment]
Hi there. We're still seeing this issue. Our workaround (setting a User Agent to anything other than the default) still works fine, but I was wondering if this been closed because you think it's fixed, or closed because it can't/won't be fixed.

**Comment by phalt** [comment]
@martinpeck HTTP clients should set a user-agent header. I know that some do not. This isn't a problem with PokéAPI :)

We're considering adding a note to documentation to help with this.

**Comment by iffanmajid** [comment]
how to set user agent for this API?

**Comment by martinpeck** [comment]
@iffanmajid exactly how you go about setting the user-agent will depend upon the language/framework/library you're using to call the API

I suspect you're actually asking "what should I set the user-agent to". Just set it to *something* other than the default/nothing. So, if you have an app called "my test app" set it to that. You can see that I set it to "cheese" in my gist (https://gist.github.com/martinpeck/2e2437bdc586a74cbc9e).

**Comment by iffanmajid** [comment]
@martinpeck I've tried using chrome to change user agent.

**Comment by Germaniac** [comment]
Doesn't work with changed user agent on python for me

**Comment by martinpeck** [comment]
@Germaniac I've just re-tried my Gist (linked above) in IDLE, using Python 3.6.2, and it works for me. Specifically, this code works fine...

```python
def this_works(num):
    url = "http://pokeapi.co/api/v1/pokemon/"+str(num)
    request = urllib.request.Request(url)

    request.add_header('User-Agent',"cheese")

    data = urllib.request.urlopen(request).read()
    return data

result1 = this_works(1)
print(result1)

```

**Comment by martinpeck** [comment]
In fact, if you want to see it working you can visit this Trinket.io page...

https://trinket.io/python3/651e9ef2b5

**Comment by Germaniac** [comment]
Your gist doesn't work on my IDLE, I wonder why...

**Comment by martinpeck** [comment]
If you post the details of your error, I might be able to help.

**Comment by Germaniac** [comment]
`Traceback (most recent call last):
  File "C:/Users/User/Downloads/poke.py", line 29, in <module>
    result1 = this_works(1)
  File "C:/Users/User/Downloads/poke.py", line 24, in this_works
    data = urllib.request.urlopen(request).read()
  File "C:\Users\User\AppData\Local\Programs\Python\Python36-32\lib\urllib\request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Users\User\AppData\Local\Programs\Python\Python36-32\lib\urllib\request.py", line 532, in open
    response = meth(req, response)
  File "C:\Users\User\AppData\Local\Programs\Python\Python36-32\lib\urllib\request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python36-32\lib\urllib\request.py", line 570, in error
    return self._call_chain(*args)
  File "C:\Users\User\AppData\Local\Programs\Python\Python36-32\lib\urllib\request.py", line 504, in _call_chain
    result = func(*args)
  File "C:\Users\User\AppData\Local\Programs\Python\Python36-32\lib\urllib\request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
>>> cvfgn
Traceback (most recent call last):
  File "<pyshell#1>", line 1, in <module>
    cvfgn
NameError: name 'cvfgn' is not defined`

**Comment by paticobr** [comment]
request.add_header('User-Agent',"cheese") works for me! 
I added ("Content-Encoding", "gzip") too.

**Comment by pbldmngz** [comment]
I've encountered this issue specifically on one of my paginated requests:
`https://pokeapi.co/api/v2/pokemon?offset=1239&limit=21` This raises 403 - Forbidden
And this `https://pokeapi.co/api/v2/pokemon?offset=1260&limit=21` is just fine.

Basically I created pages of 21 elements, which gives me around 61 pages. Page 60 is the only one that it seems that I cannot access. Is there some secret Pokemon there or why could this be happening?
